cmd/configschema/configschema/resolver.go

```diff
@@ -52,7 +57,85 @@ func NewDirResolver(srcRoot string, moduleName string) DirResolver {
 }
 
 // PackageDir accepts a Type and returns its package dir.
-func (dr DirResolver) PackageDir(t reflect.Type) string {
+// If package is not found locally, searches for package location through go.mod.
+// Attempts to search through an absolute path or GOPATH
+func (dr DirResolver) PackageDir(t reflect.Type) (string, error) {
 	pkg := strings.TrimPrefix(t.PkgPath(), dr.ModuleName+"/")
+	dir := localPackageDir(dr, pkg)
+	_, err := os.ReadDir(dir)
+	if err != nil {
+		if dir, err = externalPackageDir(dr, pkg); err != nil {
+			return "", fmt.Errorf("could not find the pkg: %s", pkg)
+		}
+	}
+	return dir, nil
+}
+
+// localPackageDir returns the path to a local package.
+func localPackageDir(dr DirResolver, pkg string) string {
 	return path.Join(dr.SrcRoot, pkg)
 }
+
+// externalPackageDir returns the path to an external package.
+func externalPackageDir(dr DirResolver, pkg string) (string, error) {
+	line, version, err := grepMod(path.Join(dr.SrcRoot, "go.mod"), pkg)
+	if err != nil {
+		return "", err
+	}
+	goPath := os.Getenv("GOPATH")
+	if goPath == "" {
+		goPath = build.Default.GOPATH
+	}
+	pkgPath := buildExternalPath(dr, goPath, pkg, line, version)
+	if _, err = os.ReadDir(pkgPath); err != nil {
+		return "", err
+	}
+	return pkgPath, nil
+}
+
+// grepMod returns the line within go.mod associated with the package
+// we are looking for.
+func grepMod(goModPath string, pkg string) (string, string, error) {
+	file, err := os.ReadFile(goModPath)
+	if err != nil {
+		return "", "", err
+	}
+	modContents, err := modfile.Parse(goModPath, file, nil)
+	if err != nil {
+		return "", "", err
+	}
+	var defaultModulePath, defaultModuleVersion string
+	for _, line := range modContents.Replace {
+		switch {
+		case strings.Compare(line.Old.Path, DefaultModule) == 0:
```

Comment:
I don't think `strings.Compare()` is a best practice:
```
// Compare is included only for symmetry with package bytes.
// It is usually clearer and always faster to use the built-in
// string comparison operators ==, <, >, and so on.
```

```suggestion
		case line.Old.Path == DefaultModule:
```